### PR TITLE
Improve GetRuntimeClient compatibility

### LINKED ISSUE #232: Support install crane-agent in Kind
**Describe the bug**

wecom-temp-cbdaadbbc7e0969eff1fb91c934eec55
install crane-agent in kind failed

**Reproduce steps**

**Expected behavior**

Crane might support install agent in kind cluster.

**Screenshots**

**Environment (please complete the following information):**
 - K8S Version: [e.g. 1.19]
 - Crane Version: [e.g. 0.1.0] 0.3.0

### PR DESCRIPTION
**Pull request by lbbniu**
#### Which issue(s) this PR fixes:

Fixes #232 #251 #276

### CODE REVIEW

File: pkg/ensurance/runtime/runtime.go [hunk 1]
```diff
@@ -10,42 +10,32 @@ import (
...
-		klog.V(2).Infof("Runtime connect using default endpoints: %v. You can set the endpoint instead.", defaultRuntimeEndpoints)
-		return gprcconnection.InitGrpcConnection(defaultRuntimeEndpoints)
+func getRuntimeClientConnection(runtimeEndpoint string) (*grpc.ClientConn, error) {
+	var runtimeEndpoints []string
+	if runtimeEndpoint != "" {
+		runtimeEndpoints = append(runtimeEndpoints, runtimeEndpoint)
 	}
-
-	return gprcconnection.InitGrpcConnection([]string{runtimeEndpoint})
+	runtimeEndpoints = append(runtimeEndpoints, defaultRuntimeEndpoints...)
```
Comment (zsnmwy) [inline pkg/ensurance/runtime/runtime.go:19]: Add the k3s runtime endpoint to defaultRuntimeEndpoints.
https://github.com/gocrane/crane/blob/76d56448d3fd23614469d722f467881a5cfa68bc/pkg/ensurance/runtime/runtime_unix.go#L6

**Comment by qmhu** [comment]
@chenkaiyue Please take a look

**Comment by chenkaiyue** [comment]
Thanks for your contributions. Have you testes in Kind or K3S, maybe you can paste test images here?

**Comment by zsnmwy** [comment]
I have some tests in `finops-docker.pkg.coding.net/gocrane/crane/crane-agent:pr-273-c744466`, `k8s v1.23.5`.

Connect container runtime endpoint successfully, but have some permission problems. 

```log
    1 connection.go:42] connect endpoint 'unix:///var/run/dockershim.sock', make sure you are running as root and the endpoint has been started
    1 connection.go:26] Connect using endpoint 'unix:///run/containerd/containerd.sock' with '2s' timeout
    1 connection.go:44] Connected successfully using endpoint: unix:///run/containerd/containerd.sock
    1 agent.go:157] Failed to get noderesource tsp configmap : configmaps "noderesource-tsp-template" is forbidden: User "system:serviceaccount:crane-system:crane-agent" cannot
    1 agent.go:202] Discover the presence of old noderesource tsp and try to contrast the changes: noderesource-demo
    1 agent.go:206] Discover the presence of old noderesource tsp and the Tsp rules have been changed: noderesource-demo
    1 request.go:936] Error in request: resource name may not be empty
    1 agent.go:210] Failed to update noderesource tsp noderesource-demo : resource name may not be empty
    1 agent.go:60] resource name may not be empty
```

**Comment by lbbniu** [comment]
k3s:
image

**Comment by qmhu** [comment]
https://github.com/gocrane/helm-charts/pull/46/files is this pr related?

**Comment by zsnmwy** [comment]
> https://github.com/gocrane/helm-charts/pull/46/files is this pr related to this change?

yes
see: https://github.com/gocrane/crane/pull/273/files#diff-f915bb96af9fbe90557cb2b8ede3eb7903ddae1297dad168074182cd3e55eb2c

**Comment by lbbniu** [comment]
> https://github.com/gocrane/helm-charts/pull/46/files is this pr related to this change?

ok

**Comment by lbbniu** [comment]
> I have some tests in `finops-docker.pkg.coding.net/gocrane/crane/crane-agent:pr-273-c744466`, `k8s v1.23.5`.
> 
> Connect container runtime endpoint successfully, but have some permission problems.
> 
> ```
>     1 connection.go:42] connect endpoint 'unix:///var/run/dockershim.sock', make sure you are running as root and the endpoint has been started
>     1 connection.go:26] Connect using endpoint 'unix:///run/containerd/containerd.sock' with '2s' timeout
>     1 connection.go:44] Connected successfully using endpoint: unix:///run/containerd/containerd.sock
>     1 agent.go:157] Failed to get noderesource tsp configmap : configmaps "noderesource-tsp-template" is forbidden: User "system:serviceaccount:crane-system:crane-agent" cannot
>     1 agent.go:202] Discover the presence of old noderesource tsp and try to contrast the changes: noderesource-demo
>     1 agent.go:206] Discover the presence of old noderesource tsp and the Tsp rules have been changed: noderesource-demo
>     1 request.go:936] Error in request: resource name may not be empty
>     1 agent.go:210] Failed to update noderesource tsp noderesource-demo : resource name may not be empty
>     1 agent.go:60] resource name may not be empty
> ```

@qmhu @chenkaiyue Should this bug be fixed by this pr or fixed by a separate pr?

**Comment by lbbniu** [comment]
kind connect container runtime endpoint successfully:
image

**Comment by chenkaiyue** [comment]
Please squash these commits.

**Comment by lbbniu** [comment]
> Please squash these commits.

Finish

**Comment by qmhu** [comment]
/LGTM

nice job!

**Pull request merged**